web/src/app/page.tsx

```diff
@@ -0,0 +1,149 @@
+'use client'
+
+import { Fragment, useEffect, useRef, useState } from 'react'
+import Spinner from '~/components/Spinner'
+import { siteConfig } from '~/config/site'
+import useStore from '~/context/useStore'
+import { cn } from '~/utils/classNames'
+import { useInView } from 'framer-motion'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import { FaDocker } from 'react-icons/fa'
+import CopyButton from '~/components/copy-button'
+import { LiaFileInvoiceSolid } from 'react-icons/lia'
+
+export default function Home() {
+  const { titleInView: t, setTitleInView, compose, code, error } = useStore()
+  const titleRef = useRef<HTMLHeadingElement>(null)
+  const titleInView = useInView(titleRef, { margin: '0px 0px -100px 0px' })
+  const [loading, setLoading] = useState(false)
+
+  useEffect(() => {
+    if (titleInView !== t) setTitleInView(titleInView)
+  }, [setTitleInView, t, titleInView])
+
+  interface CommandInput {
+    command: string
+  }
+
+  const {
+    register,
+    handleSubmit,
+    setFocus,
+    formState: { errors, isSubmitting, isValid }
+  } = useForm<CommandInput>({ mode: 'onChange', reValidateMode: 'onChange' })
+
+  useEffect(() => {
+    function onKeyDown(e: KeyboardEvent) {
+      if (
+        e.key !== '/' ||
+        (e.target as HTMLElement).tagName === 'INPUT' ||
+        (e.target as HTMLElement).tagName === 'SELECT' ||
+        (e.target as HTMLElement).tagName === 'TEXTAREA' ||
+        (e.target as HTMLElement).isContentEditable
+      ) {
+        return
+      }
+      e.preventDefault()
+      setFocus('command')
+    }
+    window.addEventListener('keydown', onKeyDown)
+    return () => window.removeEventListener('keydown', onKeyDown)
+  }, [setFocus])
+
+  useEffect(() => {
+    if (isSubmitting && isValid) setLoading(true)
+    else setLoading(false)
+  }, [isSubmitting, isValid])
+
+  const onSubmit: SubmitHandler<CommandInput> = data => compose(data.command).finally(() => setLoading(false))
+
+  return (
+    <main className={cn('')}>
+      {/* header */}
+      <header className="flex min-h-[40vh] flex-col justify-center px-5 sm:min-h-[35vh] sm:px-8 lg:px-16">
+        <div className="">
+          <div className="mx-auto max-w-lg space-y-8">
+            <h1 ref={titleRef} className="text-center text-4xl font-bold uppercase">
+              {siteConfig.name}
+            </h1>
+            <p className="text-center text-lg">{siteConfig.description}</p>
+          </div>
+        </div>
+      </header>
+      {/* main */}
+      <div className="px-5 sm:px-8 lg:px-16">
+        {/* form */}
+        <form action="#" method="post" onSubmit={handleSubmit(onSubmit)} className="">
+          <div className="mx-auto max-w-lg space-y-4 px-2">
+            {/* label */}
+            <div className={cn('flex flex-col gap-5')}>
+              <label htmlFor="command-input" className="font-bold">
+                Enter Command to Generate `<em>docker-compose.yml</em>` *
```

Comment:
Let's change this to "Paste your `docker run` command below!"

The "docker run" should be a link pointing to the reference page https://docs.docker.com/engine/reference/run/